mlflow/langchain/__init__.py

```diff
@@ -0,0 +1,385 @@
+"""
+The ``mlflow.langchain`` module provides an API for logging and loading LangChain models.
+This module exports multivariate LangChain models in the langchain flavor and univariate
+LangChain models in the pyfunc flavor:
+
+LangChain (native) format
+    This is the main flavor that can be accessed with LangChain APIs.
+:py:mod:`mlflow.pyfunc`
+    Produced for use by generic pyfunc-based deployment tools and for batch auditing
+    of historical forecasts.
+
+.. _LangChain:
+    https://python.langchain.com/en/latest/index.html
+"""
+import os
+import yaml
+import pandas as pd
+import mlflow
+import logging
+
+from mlflow.types.schema import Schema, ColSpec, DataType
+from mlflow import pyfunc
+from mlflow.environment_variables import _MLFLOW_OPENAI_TESTING
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _validate_env_arguments,
+    _process_pip_requirements,
+    _process_conda_env,
+    _CONDA_ENV_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+    _CONSTRAINTS_FILE_NAME,
+    _PYTHON_ENV_FILE_NAME,
+    _PythonEnv,
+)
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import _save_example
+from mlflow.models import Model, ModelInputExample
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.model_utils import (
+    _get_flavor_configuration,
+    _validate_and_copy_code_paths,
+    _add_code_from_conf_to_system_path,
+    _validate_and_prepare_target_save_path,
+)
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from typing import Dict, List, Union
+
+logger = logging.getLogger(mlflow.__name__)
+
+FLAVOR_NAME = "langchain"
+_MODEL_DATA_FILE_NAME = "model.yaml"
+_MODEL_DATA_KEY = "model_data"
+_MODEL_TYPE_KEY = "model_type"
+
+
+def get_default_pip_requirements():
+    """
+    :return: A list of default pip requirements for MLflow Models produced by this flavor.
+             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment
+             that, at a minimum, contains these requirements.
+    """
+    return [_get_pinned_requirement("langchain")]
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`.
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
```

Comment:
Can we add `@experimental` to all the publich APIs in this module?

```suggestion
@experimental
@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
```

Reply:
> Do we add it to huggingface or openai flavor?

We do in case we need to make breaking changes.

Reply:
sg, added